packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -363,13 +367,15 @@ - (void)picker:(PHPickerViewController *)picker
   NSNumber *maxHeight = [_arguments objectForKey:@"maxHeight"];
   NSNumber *imageQuality = [_arguments objectForKey:@"imageQuality"];
   NSNumber *desiredImageQuality = [self getDesiredImageQuality:imageQuality];
+  NSMutableArray *pathList = [NSMutableArray new];
 
   for (PHPickerResult *result in results) {
     [result.itemProvider
         loadObjectOfClass:[UIImage class]
         completionHandler:^(__kindof id<NSItemProviderReading> _Nullable image,
                             NSError *_Nullable error) {
           if ([image isKindOfClass:[UIImage class]]) {
+            dispatch_semaphore_t resultSemaphore = dispatch_semaphore_create(0);
             __block UIImage *localImage = image;
             dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
now that we handle paths in the end, we can let all these run on background and only dispatch to main queue for `[self handlePath:pathList resultCount:results.count];` below.